### PR TITLE
Oracle Automation Config

### PR DESCRIPTION
**Pull request by maneta**
Oracle Configuration Process automation for system.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+MKFILE_PATH  := $(abspath $(lastword $(MAKEFILE_LIST)))
+THISDIR_PATH := $(patsubst %/,%,$(abspath $(dir $(MKFILE_PATH))))
+PROJECT_PATH := $(patsubst %/,%,$(dir $(MKFILE_PATH)))
+.DEFAULT_GOAL := help
+
+#Default Optional Variables
+NAME ?= system-oracle
+NAMESPACE ?= 3scale-amp22
+RELEASE_VERSION ?= CR1
+BASE_IMAGE ?= brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/3scale-amp22/system:3scale-amp-2.2-rhel-7-candidate-59069-20180423135349
```
Comment (mikz) [inline Makefile]: Not public.

Comment (maneta) [reply Makefile]: Oops, I will place this as a mandatory variable, to avoid confusion.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
```
Comment (mikz) [inline Dockerfile]: I think this does not work with RHEL Docker.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    --username=${RHEL_SUB_USER} --password=${RHEL_SUB_PASSWD}
+
+RUN RHEL_MAJOR=$(set -euo pipefail; cat /etc/redhat-release | \
+      sed -E -e 's/.*release\s+([0-9]+).*/\1/') \
+ && echo "Enabling repositories for ${RHEL_MAJOR}" \
+ && yum repolist --disablerepo=\* 2> /dev/null \
+ && yum-config-manager --disable \* &> /dev/null \
+ && yum-config-manager --enable rhel-${RHEL_MAJOR}-server-rpms \
+ && echo "Installing Oracle Dependencies" \
+ && INSTALL_PKGS="libaio" \
```
Comment (mikz) [inline Dockerfile]: We can totally ship this in the base image to avoid the need to subscribe RHEL ?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ && yum install -y --setopt=tsflags=nodocs ${INSTALL_PKGS} \
+ && rpm -V ${INSTALL_PKGS} \
+ && yum clean all -y \
+ && rm -rf /var/cache/yum
+
+COPY ./oracle-client-files/instantclient-basiclite-linux.x64-12.2.*.zip /opt/oracle/
```
Comment (mikz) [inline Dockerfile]: You can COPY multiple files to same directory as one instruction.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+COPY ./oracle-client-files/instantclient-odbc-linux.x64-12.2.*.zip /opt/oracle/
+COPY ./oracle-client-files/instantclient-sdk-linux.x64-12.2.*.zip  /opt/oracle/
+
+COPY ./config/odbcinst.ini /etc/odbcinst.ini
+COPY ./tmp/odbc.ini /etc/odbc.ini
```
Comment (mikz) [inline Dockerfile]: Does this file invalidate build cache every time you build it?

Even if might be better to move it to the end to not need to unpack files when just odbc template changed.

**Comment by mikz** [comment]
I wonder why we don't provide a BuildConfig rather than this.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+ENV LD_LIBRARY_PATH=/opt/oracle/instantclient_12_2/ \
+    ORACLE_HOME=/opt/oracle/instantclient_12_2/
+
+RUN unzip /opt/oracle/instantclient-basiclite-linux.x64-12.2.0.1.0.zip -d /opt/oracle/ \
```
Comment (mikz) [inline Dockerfile]: Here you specify full path to the zip file but other parts use `*`. Better to use `*` here too.

File: Dockerfile (hunk 3 above)
Comment (mikz) [reply Dockerfile]: Last release of libaio was in 2014: https://rpmfind.net/linux/RPM/centos/7.4.1708/x86_64/Packages/libaio-0.3.109-13.el7.x86_64.html
I'd say it is safe to assume it is not high churn.

Comment (maneta) [reply Dockerfile]: +1 for me

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+# Mandatory arguments user and password for Red Hat subscription.
+ARG RHEL_SUB_USER
+ARG RHEL_SUB_PASSWD
+
+RUN (subscription-manager unregister || true) \
+ && subscription-manager clean \
+ && echo "Subscribing (takes a while)" \
+ && subscription-manager register --auto-attach \
+    --username=${RHEL_SUB_USER} --password=${RHEL_SUB_PASSWD}
```
Comment (unleashed) [inline Dockerfile]: you do not need any of this __if building under RHEL__, as RH's docker will manage subscriptions for the images based on the host's subscription.

Comment (maneta) [reply Dockerfile]: Well I'm not building under rhel, and actually whole this I have ripped of from here: https://github.com/3scale/productization/blob/master/prod-base/Dockerfile#L5-L20

File: Dockerfile (hunk 4 above)
Comment (maneta) [reply Dockerfile]: done in 87438ab2ee6b5dbe2af95273983b1e02dbbb49db

File: Dockerfile (hunk 6 above)
Comment (maneta) [reply Dockerfile]: done in 87438ab2ee6b5dbe2af95273983b1e02dbbb49db

File: Dockerfile (hunk 2 above)
Comment (maneta) [reply Dockerfile]: hmm, really not sure about it. I would need to review the documentation

File: Dockerfile (hunk 3 above)
Comment (maneta) [reply Dockerfile]: @mikz @hallelujah if we are willing to ship `libaio` we need to decide fast, so @ddcesare could ask for the package authorization in the build.

File: Dockerfile (hunk 7 above)
Comment (mikz) [reply Dockerfile]: @maneta but our customers will be building under RHEL, right? And they'll probably not be ok with using subscriptions to this and exposing user/password in such way.

File: config/odbc.ini.tpl [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+MaxLargeData = 0
+MetadataIdDefault = F
+QueryTimeout = T
+ResultSets = T
+ServerName = threescale
+SQLGetData extensions = F
+Translation DLL =
+Translation Option = 0
+DisableRULEHint = T
+UserID = $ORACLE_UID
```
Comment (maneta) [inline config/odbc.ini.tpl]: @mikz I think that a BuildConfig is a good doable idea, the only trick part (at least for me) would setting the Oracle `UserID` and `Password`

File: Dockerfile (hunk 2 above)
Comment (mikz) [reply Dockerfile]: I can save you the time. It does not work.

```
ARG BASE_IMAGE
FROM ${BASE_IMAGE}
RUN env
```

```shell
docker build --build-arg BASE_IMAGE=centos:7 .
```

```
Sending build context to Docker daemon 2.048 kB
Step 1/3 : ARG BASE_IMAGE
Please provide a source image with `from` prior to commit
```

File: Dockerfile (hunk 7 above)
Comment (maneta) [reply Dockerfile]: Ok so we are removing this whole stuff as we are productizing the image with libaio

**Comment by maneta** [comment]
good to se you here guys, we are closing this and work a `BuildConfig`

**Pull request closed without merging**